### ISSUE TITLE
Quarto vignette titles are broken with quarto 1.6.x

### ISSUE THREAD

**Issue by banfai** [r-lib/pkgdown#2823]
With the pre-release version of quarto the page title has a glitch, it's in a different div `<header id="title-block-header" class="quarto-title-block">)` instead of the previous `<div class="page-header">`. This causes the Source link to be placed above the title:

release version on `gh-pages`:
<img width="774" alt="quarto-vignette-1 5 x" src="https://github.com/user-attachments/assets/dd3f4093-f466-4dcd-82b9-f510d1c73270">

same version rendered with quarto 1.6.37:
<img width="773" alt="quarto-vignette-1 6 37" src="https://github.com/user-attachments/assets/97bfeb64-299c-4650-91c7-2b875429f989">

The `page-header` div is still there with the logo and the source code link, but the title is missing: `<h1></h1>`

```r
quarto::quarto_version()
#> [1] '1.6.37'
```

It remains the same with specifying
```yaml
format:
  html:
    title-block-style: none
```

**Comment by banfai** [comment]
it might be caused by this:
https://github.com/quarto-dev/quarto-cli/issues/10567
https://github.com/quarto-dev/quarto-cli/pull/11224

**Comment by jayhesselberth** [comment]
Quarto tests are also failing with the new release:

```
Failure (test-build-quarto-articles.R:59:3): auto-adjusts heading levels
xpath_text(html, "//h1") (`actual`) not equal to "title" (`expected`).

`actual`:   "" "title"
`expected`:    "title"

Failure (test-build-quarto-articles.R:104:3): quarto headings get anchors
xpath_attr(headings, "./a", "href") (`actual`) not equal to c("#heading-1", "#heading-2") (`expected`).

`actual`:   "#heading-1"             
`expected`: "#heading-1" "#heading-2"
```

**Comment by banfai** [comment]
> It's not clear from the comments whether the change in [quarto-dev/quarto-cli#11224](https://github.com/quarto-dev/quarto-cli/pull/11224) will be permanent. Perhaps we should clarify before updating pkgdown to process the changed HTML.

should I open an issue for quarto?

> I'm assuming we don't want to be supporting multiple versions of quarto, so if we update to process the new HTML, we'd need to bump the required quarto version to 1.6.33.

this might be an issue for some time since 1.6.x is still pre-release

**Comment by cderv** [comment]
It took me some time to understand the problem and this is indeed quarto-dev/quarto-cli#11224 having side effect on the custom processing that **pkgdown** is doing

Within pkgdown, `quarto render` is called with a custom template, that create a `div` structure with some information 
https://github.com/r-lib/pkgdown/blob/dfa28bef27858bd442334b5cce8f645d555914be/inst/quarto/template.html#L25-L48

The custom template is applying which leads to this in the HTML **before** Quarto postprocessing
````html
<div class="meta">
  <h1>My cool title</h1>
</div>
````

However, as the custom template does not use `title-block.html` partial, it will trigger the post processing introduced by PR quarto-dev/quarto-cli#11224 as if the document had no title. 
https://github.com/quarto-dev/quarto-cli/blob/39baf4081f184fe090974295b48d1beab32cd1e4/src/format/html/format-html-title.ts#L205

and a title block will be created anyway by this new post-processing. The `h1` from the custom template is caught
https://github.com/quarto-dev/quarto-cli/blob/39baf4081f184fe090974295b48d1beab32cd1e4/src/format/html/format-html-title.ts#L210
and the `h1` that **pkgdown** expect will be moved 
https://github.com/quarto-dev/quarto-cli/blob/39baf4081f184fe090974295b48d1beab32cd1e4/src/format/html/format-html-title.ts#L238

I don't know exactly why **pkgdown** does this title processing with specific custom template, but this is indeed a Quarto issue in the sense that it should not probably apply this eagerly

**Comment by banfai** [comment]
> @banfai
> 
> > should I open an issue for quarto?
> 
> I'll do it
thanks!

> 
> > this might be an issue for some time since 1.6.x is still pre-release
> 
> Quarto 1.6 is the latest release since last week.
oops, I missed that

**Comment by hadley** [comment]
Can we make pkgdown work either way? This is causing pkgdown to fail on CRAN and it's likely easier to submit an updated pkgdown than get CRAN to update their quarto.

**Comment by jayhesselberth** [comment]
@hadley we don't need CRAN to update the quarto package, the quarto package uses system quarto (you have to install the quarto CLI to use the R package). I see CRAN wants this fixed by Dec 24, how festive.

We could fix this to work with quarto 1.6, but may then have to revert if this is "fixed" on the quarto CLI side.

**Comment by jayhesselberth** [comment]
I am also not clear what specific quarto CRAN is installing in their systems for checks. Is that documented somewhere?

**Comment by jayhesselberth** [comment]
Quarto devs are planning for a quick fix https://github.com/quarto-dev/quarto-cli/pull/11224#issuecomment-2523338061 so maybe we sit tight.

**Comment by cderv** [comment]
Yes we have a fix that should sove the problem for pkgdown. We'll do a patch release. 

> I am also not clear what specific quarto version that CRAN is installing in their systems for checks. Is that documented somewhere?

Though I am like you and I don't know for sure if CRAN update Quarto at each stable release we do

**Comment by jayhesselberth** [comment]
Thanks @cderv for the quick fix!

**Comment by jayhesselberth** [comment]
This should be fixed now, see #2828

**Comment by cderv** [comment]
Glad it works ok now. I don't know if CRAN will update their version, but if they do follow release then they may do it. We'll see.
